src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -46,30 +97,30 @@ public void removeSection(Section section) {
 		return stations.collect(Collectors.toList());
 	}
 
-	private Stream<Station> getStationStream(Section section) {
-		return Stream.of(section.getUpStation(), section.getDownStation());
+	private Stream<Station> getStationStream(Optional<Section> section) {
+		return section.map(value -> Stream.of(value.getUpStation(), value.getDownStation())).orElseGet(Stream::empty);
 	}
 
-	private boolean hasNext(Section currentSection) {
-		return this.sections.stream()
-			.anyMatch(section -> section.getUpStation().equals(currentSection.getDownStation()));
+	private boolean hasNext(Optional<Section> currentSection) {
```

Comment:
Optional을 인스턴스 변수나 파라미터로 활용하는 부분은 권장되지 않습니다.

Section을 파라미터로 넘기는건 어떨까요?


```
Item 16: Do Not Use Optional in Methods Arguments
Don't force call sites to create Optionals. Do not use Optional as fields or in setters and constructors' arguments.

UsingOptionalin methods arguments is another common mistake. This would lead to code that is unnecessarily complicated. Take responsibility for checking arguments instead of force call sites to create Optionals. This practice clutters the code and may cause dependence. In time, you will use it everywhere. Keep in mind thatOptionalis just another object (a container) and is not cheap — it consumes 4x the memory of a bare reference! But you may also like to check this out and decide, depending on the case.
```

https://dzone.com/articles/using-optional-correctly-is-not-optional

Reply:
참고 링크 감사합니다..👍🏻 

Optional을 반환하다보니 결국 check해야하는 로직이 생기고 indent가 늘어나게 되어 Optional 파라미터를 넘기게 된건데 성능과 의존성에 문제가 있었네요..

처음 의견주신 null반환 하는 방식으로 수정했습니다.